code/game/objects/structures/mineral_doors.dm

```diff
@@ -83,26 +83,30 @@
 		icon_state = "[base_icon_state][smoothing_flags ? "-[smoothing_junction]" : ""]"
 
 
-/obj/structure/mineral_door/attackby(obj/item/W, mob/living/user)
+/obj/structure/mineral_door/attackby(obj/item/attacking_item, mob/living/user)
 	. = ..()
 	if(.)
 		return
 	if(QDELETED(src))
 		return
 
+	calculate_attackby(attacking_item, user)
```

Comment:
the way this is structured, this won't be called if you ARE in harm intent for non resin type mineral walls.